fairroot/base/sim/FairMCApplication.cxx

```diff
@@ -926,9 +918,29 @@ void FairMCApplication::InitGeometry()
 
     fGeometryIsInitialized = kTRUE;
 
+    if (fIonsAreAdded && fGeometryIsInitialized) {
+        InitFinalizer();
+    }
+
     fState = FairMCApplicationState::kUnknownState;
 }
 
+//_____________________________________________________________________________
+void FairMCApplication::InitFinalizer()
+{
+    std::lock_guard guard(mtx);
```

Comment:
Why the `lock_guard`?

If you think that `fRootManager->WriteFolder();`needs it, then please only put that one under mutex control?

```cpp
{
    std::lock_guard guard(mtx);
    fRootManager->WriteFolder();
}
```